doc/api/test.md

```diff
@@ -795,7 +801,7 @@ changes:
 * `options` {Object} Configuration options for the test. The following
   properties are supported:
   * `concurrency` {number|boolean} If a number is provided,
-    then that many tests would run in parallel.
+    then that many tests would run in parallel within the application thread.
```

Comment:
Maybe this adds clarity?

```suggestion
    then that many tests would run in parallel within the current thread, similar to a set of promises awaited by `Promise.allSettled`.
```

Reply:
I don't think using `current` vs `application` does a great job at clarifying things. Why do you think it's important to call out `Promise.allSettled` vs anything else? I don't think it's helpful, it might even be a bit confusing given that the code doesn't use `Promise.allSettled`.

Reply:
I’m trying to get across that it’s “in parallel” only in the sense that there are several async resources running within the current thread and process, as opposed to “true” parallelism from using multiple threads/processes.